### ISSUE TITLE
What should be on the user profile?

### ISSUE THREAD

**Issue by mattl** [foocorp/hacienda#14]
Today I got basic user account management working. 

You can see a work in progress screenshot on my Mastodon profile: https://social.coop/@mattl/111875224875770905

I have now taken this a stage further and got the very beginning of a working profile page. I say "working" -- it's more of a "here's the user's name" and a Gravatar and then a list of all the fields we're collecting on the user management page, including passwords. We're not going to show your password and in fact,  your password will be encrypted, so all I will see in the database is $2y$10$J8KaDyRRwPxQWa/upe9lo.JBY0CdrSeGgSx5mAmSm99P9oNsHC09S (a bcrypt of `hunter2` in this case)

<img width="1159" alt="Screenshot 2024-02-04 at 6 36 08 PM" src="https://github.com/foocorp/hacienda/assets/33296/6f9e9035-9134-42de-9ce5-f956cb6d73dc">

What should be on the user profile? 

https://libre.fm/user/mattl

Of all the pages on the current site, I dislike the user profile the least right now. I think we can move the free artists stuff to be better organized. 

What are some of your ideas?

**Comment by evnb** [comment]
Idk exactly what scrobble data you have access to so just throwing some ideas out there:
- total scrobble-hours (with per-service breakdown) plus visualization maybe GitHub style or just line graph
- total scrobble song plays per streaming platforms (non-unique)
- total unique songs/albums/artists both per and across streaming platforms
- user’s favorite playlists
- links to user’s accounts on streaming platforms

**Comment by virtadpt** [comment]
"Encrypted" or "hashed"?

Can you make Gravatar optional?  Not everybody trusts a single service for their avatar.  Being able to upload one of one's own would be more self-hosty and less "this service decided to end what they do so enjoy watching your profile take two minutes to time out."

**Comment by mattl** [comment]
bcrypt hashed+salted.

Not interested in hosting avatars... I'd rather stick with Gravatar for now. My primary focus here is building this for Libre.fm not for others to self-host. If people do wind up doing that, then someone else might want to add that functionality later?

**Comment by mattl** [comment]
Seems like libravatar.org is a viable alternative, and we may get that working. I'll open up another issue for it.

**Comment by janerationx** [comment]
Can we go back to square avatars?

**Comment by jbwharris** [comment]
I know myself when it comes to a profile on LastFM I wish there were ways to highlight different metrics. So for someone like me, the impressive stats aren't the number of scrobbles per artist, but the amount of artists scrobbled, or total different tracks. In the same way someone else might want to highlight albums, or total scrobbles.

**Comment by mattl** [comment]
> Can we go back to square avatars?

As someone with a round one, yes we can.

**Comment by inan-hira** [comment]
Would be nice to have the ability to add `rel="me"` to the social links, for [verification on Mastodon][m] for instance

[m]: https://docs.joinmastodon.org/user/profile/#verification

**Comment by mattl** [comment]
@inan-hira Done!

@janerationx Done!

**Comment by 646a70** [comment]
> Not interested in hosting avatars... I'd rather stick with Gravatar for now. My primary focus here is building this for Libre.fm not for others to self-host. If people do wind up doing that, then someone else might want to add that functionality later?

Is this why I can't find the 'Avatar address' field on the 'Edit' page? Under 'Help' it currently states "The web address for a picture to represent you on Libre.fm. It should not be more than 80x80 pixels. (64x64 is best.) If you leave this empty, we'll use Libravatar.org to find an image for you.". Also, I can't get the 'Fediverse Profile' field to record my Mastodon profile address.

**Comment by mattl** [comment]
Yes it’s only using Libravatar.org now…. Almost nobody had a custom avatar URL. 

Your fediverse URL should save it may just take a while to appear on your profile. What’s your Libre.fm profile URL?

**Comment by 646a70** [comment]
> Your fediverse URL should save it may just take a while to appear on your profile. What’s your Libre.fm profile URL?

https://libre.fm/user/djp

**Comment by poleguy** [comment]
In response to your fediverse message. My message got garbled by mastodon... This is what I tried to write:

Also the "Fediverse Profile" help says "address for your profile" which I interpreted as "@poleguy@mastodon.social". Maybe you should say "The URL of your profile on the social web on an ActivityPub-powered site, e.g. "https://social.lol/@username"

Although maybe not that url, because it's somebody's real url. :-)

Unfortunately I did not know mastodon would "shorten" my unique username. So you didn't get my intent.

@poleguy@mastodon.social is unique. My suggestion is that if you require the url format for the profile page you should note that clearly. To avoid ambiguity.

**Comment by mattl** [comment]
The field is now a URL type and has a more obvious example by default. @poleguy we need to get you scrobbling stuff now.

**Comment by nathansgithub** [comment]
I haven't used libre.fm in its previous form, but this is looking good! 

Besides a top 10/whatever artists and albums (potentially with album artwork), another good feature could be the ability to hide recent plays and/or hide the time next to the list.

**Comment by alxtrnr** [comment]
Can an option to change the username be included? I set mine up years ago and would like to change it.
